### DISCUSSION TITLE
Why am I not getting completions from grouped backends?

### DISCUSSION THREAD

**Discussion by mtwomey** [company-mode/company-mode#1149] [Q&A] [answered]
#### Output of the command `M-x company-diag`
```
Emacs 27.2 (x86_64-apple-darwin19.6.0) of 2021-06-21 on MacBook-Pro.local
Company 0.9.13

company-backends: (company-capf
 (:separate company-dabbrev company-yasnippet company-ispell))

Used backend: (:separate company-dabbrev company-yasnippet company-ispell)

Major mode: org-mode
Prefix: "titl"
Completions:
  "title"
  #("titlark" 0 7 (company-backend company-ispell))
  #("title" 0 5 (company-backend company-ispell))
  #("titleboard" 0 10 (company-backend company-ispell))
  #("titled" 0 6 (company-backend company-ispell))
  #("titledom" 0 8 (company-backend company-ispell))
  #("titleholder" 0 11 (company-backend company-ispell))
  #("titleless" 0 9 (company-backend company-ispell))
  #("titleproof" 0 10 (company-backend company-ispell))
  #("titler" 0 6 (company-backend company-ispell))
  #("titleship" 0 9 (company-backend company-ispell))
  #("titlike" 0 7 (company-backend company-ispell))
  #("titling" 0 7 (company-backend company-ispell))
  #("titlist" 0 7 (company-backend company-ispell))
```

#### Describe the issue
I'm having trouble getting yasnippet completions, when there are other completions also happening. I am unsure if I'm just doing something wrong with my setup or if there is a problem here.

In this example I have a snippet with the key `%org-title-general`. If I type `%org-titl` I'll get completions from company-ispell, nothing from yasnippet. On the other hand if I _call_ `company-yasnippet` at the same point, I do get the completions.

#### Steps to reproduce
1. Set backend as shows in the debug above
1. Create a yasnippet as described
1. Try the completion from simply typing `%org-titl`
1. Try the completion by calling `company-yasnippet` directly at point

#### Expected behavior
Get the completions from both.

#### Screenshots
Completion that comes up automatically just from typing:
![image](https://user-images.githubusercontent.com/7715262/126816421-a3249231-47d4-4faa-b434-8ee0b8dc6257.png)

Completion from calling `company-yasnippet`:
![image](https://user-images.githubusercontent.com/7715262/126816504-ca63d86e-5598-4d80-b91c-d246be37d361.png)

**Comment by dgutov** [comment, marked as answer]
Hi!

This is basically the same as #792.

Unfortunately, we don't support grouping of entirely arbitrary backends, they need to be compatible in what `prefix` returns.

The default configuration is done with that aspect in mind, where `company-dabbrev-code` is grouped with `company-etags` and `company-dabbrev` stands to the side. But the latter can be combined with `company-ispell`. `company-yasnippet` works better when combined with any of the former group (they determine the prefix bounds based on "symbol" syntax).

In general this has worked okay because you usually want some backends in text buffers, and others in programming buffers, but it seems `company-yasnippet` can be attractive in both.

There is a rough idea of how to fix this in the linked issue. I don't have the time for this now, so someone is welcome to beat me to it.

**Reply by mtwomey** [reply]
Ok, understand. Thanks for the detail - I appreciate it.

**Comment by dgutov** [comment]
FWIW, if those are your personal snippets, using the `<` as the initial character should work better than `%`.

Now that #1268 has been fixed, that is.

**Comment by dgutov** [comment]
This shouldn't be a problem anymore with Company 1.0.
